vochain/indexer/indexer.go

```diff
@@ -718,7 +727,63 @@ func (idx *Indexer) OnProcessesStart(pids [][]byte) {
 
 // NOT USED but required for implementing the vochain.EventListener interface
 func (idx *Indexer) OnSetAccount(addr []byte, account *state.Account) {}
-func (idx *Indexer) OnTransferTokens(from, to []byte, amount uint64)  {}
+func (idx *Indexer) OnTransferTokens(tx *vochaintx.TokenTransfer) {
+	idx.lockPool.Lock()
+	defer idx.lockPool.Unlock()
+	idx.tokenTransferPool = append(idx.tokenTransferPool, &indexertypes.TokenTransferMeta{
+		From:      tx.FromAddress.Bytes(),
+		To:        tx.ToAddress.Bytes(),
+		Amount:    tx.Amount,
+		Height:    tx.Height,
+		TxHash:    tx.TxHash,
+		Timestamp: time.Now(),
+	})
+}
+
+// newTokenTransfer creates a new token transfer and stores it in the database
+func (idx *Indexer) newTokenTransfer(tt *indexertypes.TokenTransferMeta) error {
+	queries, ctx, cancel := idx.timeoutQueries()
+	defer cancel()
+	if _, err := queries.CreateTokenTransfer(ctx, indexerdb.CreateTokenTransferParams{
+		TxHash:       tt.TxHash,
+		Height:       int64(tt.Height),
+		FromAccount:  tt.From,
+		ToAccount:    tt.To,
+		Amount:       int64(tt.Amount),
+		TransferTime: tt.Timestamp,
+	}); err != nil {
+		return err
+	}
+	log.Debugf("adding new token transfer from %x to %x: %d into the indexer database", tt.From, tt.To, tt.Amount)
```

Comment:
`log.Debugw` will look nicer :) This is a clear example on where this function is useful.

`log.Debugw("new token transfer", map[string]interface{}{ "from": fmt.Sprintf("%x, tt.From"), "amount": tt.Amount, etc...)`